source/dub/recipe/sdl.d

```diff
@@ -343,64 +342,79 @@ private string expandPackageName(string name, string parent_name, Tag tag)
 
 private string stringTagValue(Tag t, bool allow_child_tags = false)
 {
-	import std.string : format;
 	enforceSDL(t.values.length > 0, format("Missing string value for '%s'.", t.fullName), t);
 	enforceSDL(t.values.length == 1, format("Expected only one value for '%s'.", t.fullName), t);
-	enforceSDL(t.values[0].peek!string !is null, format("Expected value of type string for '%s'.", t.fullName), t);
 	enforceSDL(allow_child_tags || t.tags.length == 0, format("No child tags allowed for '%s'.", t.fullName), t);
 	// Q: should attributes be disallowed, or just ignored for forward compatibility reasons?
 	//enforceSDL(t.attributes.length == 0, format("No attributes allowed for '%s'.", t.fullName), t);
-	return t.values[0].get!string;
+	return t.values[0].expect!string(t);
+}
+
+private T expect(T)(
+	Value value,
+	Tag errorInfo,
+	string customFieldName = null,
+	string file = __FILE__,
+	int line = __LINE__
+)
+{
+	return value.match!(
+		(T v) => v,
+		(fallback)
+		{
+			enforceSDL(false, format("Expected value of type " ~ T.stringof ~ " for '%s', but got %s.",
+				customFieldName.length ? customFieldName : errorInfo.fullName,
+				typeof(fallback).stringof),
+				errorInfo, file, line);
```

Comment:
We don't pass file/line info for the actual parsed file anywhere ?

Reply:
no